src/OmniSharp/MSBuild/ProjectFile/ProjectFileInfo.cs

```diff
@@ -175,5 +180,24 @@ public static ProjectFileInfo Create(MSBuildOptions options, ILogger logger, str
 #endif
             return projectFileInfo;
         }
+        private static LanguageVersion ToLanguageVersion(string langVersionPropertyValue)
+        {
+            if (!(string.IsNullOrWhiteSpace(langVersionPropertyValue) || langVersionPropertyValue.Equals("Default", StringComparison.OrdinalIgnoreCase)))
+            {
+                // ISO-1, ISO-2, 3, 4, 5, 6 or Default
+                switch (langVersionPropertyValue.ToLower())
+                {
+                    case "ISO-1": return LanguageVersion.CSharp1;
+                    case "ISO-2": return LanguageVersion.CSharp2;
+                    case "3": return LanguageVersion.CSharp3;
+                    case "4": return LanguageVersion.CSharp4;
+                    case "5": return LanguageVersion.CSharp5;
+                    case "6": return LanguageVersion.CSharp6;
+                }
+            }
+            return LanguageVersion.CSharp5;
```

Comment:
Just guessed that for most existing projects C# 5 is the default. Only
VS2015 may default to C# 6. The right thing probably would be to see which
Framework is being targeted but I didn't want to make it more complex.

Another alternative would be to have a Nullable<LanguageVersion> property
and let Roslyn choose the default if not specified in the csproj, by not
setting parsing options as was the code before.

Rafael Teixeira
O..:.)oooo

On Thu, Apr 30, 2015 at 7:36 PM, David Fowler <notifications@github.com>
wrote:

> In src/OmniSharp/MSBuild/ProjectFile/ProjectFileInfo.cs
> <https://github.com/OmniSharp/omnisharp-roslyn/pull/201#discussion_r29477202>
> :
>
> > +        private static LanguageVersion ToLanguageVersion(string langVersionPropertyValue)
> > +        {
> > +            if (!(string.IsNullOrWhiteSpace(langVersionPropertyValue) || langVersionPropertyValue.Equals("Default", StringComparison.OrdinalIgnoreCase)))
> > +            {
> > +                // ISO-1, ISO-2, 3, 4, 5, 6 or Default
> > +                switch (langVersionPropertyValue.ToLower())
> > +                {
> > +                    case "ISO-1": return LanguageVersion.CSharp1;
> > +                    case "ISO-2": return LanguageVersion.CSharp2;
> > +                    case "3": return LanguageVersion.CSharp3;
> > +                    case "4": return LanguageVersion.CSharp4;
> > +                    case "5": return LanguageVersion.CSharp5;
> > +                    case "6": return LanguageVersion.CSharp6;
> > +                }
> > +            }
> > +            return LanguageVersion.CSharp5;
>
> Why aren't we defaulting to C# 6? Is this the regular msbuild behavior?
> Have you looked at the MSbuildWorkspace in the roslyn code base?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OmniSharp/omnisharp-roslyn/pull/201/files#r29477202>.
>